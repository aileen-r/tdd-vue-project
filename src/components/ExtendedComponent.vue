<template>
  <b-jumbotron header="Extending VueJs Components" :lead="$options.leadInText">
    <p>{{ paragraph }}</p>
    <b-button class="github-button" variant="primary" :href="linkUrl">{{
      linkText
    }}</b-button>
  </b-jumbotron>
</template>

<script>
import BaseComponent from './BaseComponent';

export default {
  name: 'ExtendedComponent',

  extends: BaseComponent,

  computed: {
    linkText() {
      return this.linkUrl.split('/').slice(-1)[0];
    }
  }
};
</script>
