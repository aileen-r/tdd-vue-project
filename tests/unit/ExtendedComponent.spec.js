import { shallowMount } from '@vue/test-utils';

import ExtendedComponent from '@/components/ExtendedComponent';

describe('ExtendedComponent', () => {
  let props;

  // build function suggested over beforeEach() method
  const build = () => {
    const wrapper = shallowMount(ExtendedComponent, {
      propsData: props
    });

  return {
      wrapper,
      githubButton: () => wrapper.find('.github-button')
    };
  };

  beforeEach(() => {
    props = { linkUrl: 'https://url.com/hello' };
  });

  // does the component render
  it('renders the component', () => {
    // arrange
    const { wrapper } = build();
    // assert
    expect(wrapper.html()).toMatchSnapshot();
  });

  // does the component render the right thing
  it('renders main child elements', () => {
    // arrange
    const { githubButton } = build();
    // assert
    expect(githubButton().exists()).toBe(true);
  });

  it('splits url prop correctly', () => {
    const { githubButton } = build();
    expect(githubButton().html()).toContain('hello');
    // there should be a better way to do this
  });


  

  // // it binds all the correct prop data
  // it('passes a user prop to user profile component', () => {
  //   // arrange
  //   state.user = userFixture;
  //   const { userProfile } = build();
  //   // assert
  //   expect(userProfile().vm.user).toBe(state.user);
  // });
});