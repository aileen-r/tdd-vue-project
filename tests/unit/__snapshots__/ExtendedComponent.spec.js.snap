// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtendedComponent renders the component 1`] = `
<b-jumbotron-stub header="Extending VueJs Components" lead="This lead-in text is an $option from the base component">
  <p>This paragraph text is a computed prop from the base component. The GitHub URL below is passed as a prop to the base component. The link text is a computed prop on the extended. This "linkText" computed prop accesses a prop on the base component via the "this" keyword.</p>
  <b-button-stub variant="primary" href="https://url.com/hello" class="github-button">hello</b-button-stub>
</b-jumbotron-stub>
`;
